### PR TITLE
Fix #1598 : Issues when mentioning a user

### LINKED ISSUE #1598: Issues when mentioning a user
**Describe the bug**

I type "Hi @m", tap on @michael, press space, and "ichael" disappears and the keyboard closes, presumably from losing focus, I'm left with just "Hi @m". If I then tap in the editor, the cursor is back, if I tap on @m the popup shows and I can tap on @michael again. ONLY THEN can I resume typing. 

https://drive.google.com/file/d/16yDYsZWBLk98CjivdT7iW3rqoP34iRTL/view?usp=sharing

I type "@m" the popup shows up, but I decide to type something else first, so hit backspace until it's gone. Then type "Hi @m", tap on @michael and now the editor shows "Hi @m@michael". Also, I can't continue typing and a tap in the editor removes the autocompleted bit "ichael".

https://drive.google.com/file/d/19P1-Ex8QjjS1xb_Gq5Um241dzs-tdK1z/view?usp=sharing

On desktop, there are similar bugs as soon as I use backspace to remove a mention (or try and fail as the cursor gets stuck) and then try to insert the mention again. Try a new update, type "@m" select @michael, then backspace until it's gone, type "Hi @m" select @michael, and watch the text "Hi " disappear and more shenanigans as you go on and move backward through the text.

https://drive.google.com/file/d/1lg5uko-i7frWBOZQTqtmX6K3tcbiL0B5/view?usp=sharing

**Support ticket links**
https://secure.helpscout.net/conversation/1296465630/101277

### PR DESCRIPTION
**Pull request by buntybuddyboss**
Related Issue #1598 

**Fixed - Mozilla - Desktop**

>On desktop, there are similar bugs as soon as I use backspace to remove a mention (or try and fail as the cursor gets stuck) and then try to insert the mention again. Try a new update, type "@m" select @michael, then backspace until it's gone, type "Hi @m" select @michael, and watch the text "Hi " disappear and more shenanigans as you go on and move backward through the text.
https://drive.google.com/file/d/1lg5uko-i7frWBOZQTqtmX6K3tcbiL0B5/view?usp=sharing

Proof: https://www.loom.com/share/dd3c21ad8c3345a1a069e62efc580a37

================

**Fixed - Chrome - Android**

>I type "@m" the popup shows up, but I decide to type something else first, so hit backspace until it's gone. Then type "Hi @m", tap on @michael and now the editor shows "Hi @m@michael". Also, I can't continue typing and a tap in the editor removes the autocompleted bit "ichael".
https://drive.google.com/file/d/19P1-Ex8QjjS1xb_Gq5Um241dzs-tdK1z/view?usp=sharing

Proof: https://www.loom.com/share/e8fe54cb4a064d45b17118a9d72f2d02

================

**Not Fixed - Mozilla - Android**

>I type "Hi @m", tap on @michael, press space, and "ichael" disappears and the keyboard closes, presumably from losing focus, I'm left with just "Hi @m". If I then tap in the editor, the cursor is back, if I tap on @m the popup shows and I can tap on @michael again. ONLY THEN can I resume typing.
https://drive.google.com/file/d/16yDYsZWBLk98CjivdT7iW3rqoP34iRTL/view?usp=sharing

My Findings on this

- When we hit space after mention, it's removing the mentioned name.
- It's hard to debug as developer tool is not available for Mozilla Androind in browser stack.
- But one reason I found ( Not 100% sure ), it's initiating the focus from https://github.com/buddyboss/buddyboss-platform/blob/release/src/bp-core/js/mentions.js#L324 and removing the mentioned name.
- On other platform, it's not initiating the focus when we hit space.

How i was debugging

- As developer tool is not available, we can not do `console.log`.
- So I used following function:

```
function download(filename, text) {
  var element = document.createElement('a');
  element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(text));
  element.setAttribute('download', filename);

  element.style.display = 'none';
  document.body.appendChild(element);

  element.click();

  document.body.removeChild(element);
} 

```

- Used as `download('filter.json',results);`. This is equalant to `console.log(results);`.
- Difference is, in mozilla android, it will donwload a file and in that file you will have the result of what you are debugging.

================

**Task progress video with proper UPDATE:**
https://www.loom.com/share/28ca9013780f41aa9863415c2efba023

### CODE REVIEW

**Comment by jitendrabanjara1991** [comment]
Hello @buntybuddyboss @sourabhmatolia @chetansatasiya 

I have fixed the issue with space and backspace hits in Mozilla firefox on the android device. I also checked the above three issues in the browser stack and lambda test tool.

Browser Stack -  [https://screencast-o-matic.com/watch/crVrq69y3u](https://screencast-o-matic.com/watch/crVrq69y3u)
Lambda Test - [https://screencast-o-matic.com/watch/crVrqc9yFv](https://screencast-o-matic.com/watch/crVrqc9yFv)

**Review by chetansatasiya — DISMISSED** [review]
Looks good to me @jitendrabanjara1991 @buntybuddyboss 

Need 2nd opinion from @mehulkaklotar  to look the changes.

@jitendrabanjara1991 Merge in dev and add in testing.

@sumitbuddyboss OR @mohit-buddyboss Make sure test all the devices to not break anything.

**Comment by jitendrabanjara1991** [comment]
Hello @chetansatasiya 
Merged in dev.
Thanks

**Comment by sumitbuddyboss** [comment]
Hello @buntybuddyboss @jitendrabanjara1991 

Issue >
When we type @username and then click on it then the username disappears. Please check the below issue video.

Browser specification:
**Windows**
    Chrome 88.0.4324.182
    Firefox 85.0.2 

**MacOS**
 Safari 10.1
Firefox 85.0

**iPhone 12** 
Safri and Chrome 

**Android** 
Chrome and Firefox

Issue video:
https://www.loom.com/share/ff83d64847d441db89dd036b467e2557

**Comment by jitendrabanjara1991** [comment]
Hello @chetansatasiya 
@buntybuddyboss 
I have fixed the issue that @sumitbuddyboss had given, so please review it once. After your confirmation I will merge it with Dev.
Thanks.

File: src/bp-core/js/mentions.js [hunk 1]
```diff
@@ -321,13 +361,18 @@ window.bp = window.bp || {};
...
 				'focus',
 				BP_Mentions_Options.selectors.join( ',' ) ,
 				function() {
-					if ( $( this ).data( 'bp_mentions_activated' ) ) {
-						return;
-					}
+					// Issue - When we type @username and then click on it then the username disappears.
+					// if ( $( this ).data( 'bp_mentions_activated' ) ) {
+					// 	return;
+					// }
```
Comment (chetansatasiya) [inline src/bp-core/js/mentions.js:375]: Remove commented code.

File: src/bp-core/js/mentions.js [hunk 2]
```diff
@@ -278,14 +278,24 @@ window.bp = window.bp || {};
...
-						jQuery(this).removeAttr('contenteditable');
-					});
+				if (!isAndroid()) {
+					if(e.keyCode == 8){
+
+						jQuery(this).find('.atwho-inserted').each(function (  ){
+							jQuery(this).removeAttr('contenteditable');
+						});
+
+					}else{
```
Comment (chetansatasiya) [inline src/bp-core/js/mentions.js:290]: follow proper spacing

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@jitendrabanjara1991 do the requested changes

**Comment by jitendrabanjara1991** [comment]
Hello @chetansatasiya 
Changes done as per your suggestion. Please check it and let me know if anything else.
Thanks.

**Review by mehulkaklotar — COMMENTED** [review]
This changes will not work for forums when text formatting option is disabled and only textarea is there in the topic and reply form when submitting. so please check those as well.

**Comment by jitendrabanjara1991** [comment]
Hello @mehulkaklotar 
**Fixed** - This changes will not work for forums when text formatting option is disabled and only textarea is there in the topic and reply form when submitting
Here is the video - https://screencast-o-matic.com/watch/creDbkVcJtr
Please check it and let me know if anything else.
Thanks

**Review by mehulkaklotar — DISMISSED** [review]
@jitendrabanjara1991 please fix ci errors https://travis-ci.org/github/buddyboss/buddyboss-platform/builds/763152892?utm_source=github_status&utm_medium=notification

and then put this into testing.

**Comment by jitendrabanjara1991** [comment]
Hello @mehulkaklotar 
Fixed ci errors and Merged in dev
Thanks

**Comment by sumitbuddyboss** [comment]
Hello @jitendrabanjara1991 

**Issue - 1**
When we mention user (select user from up/down arrow from the keyboard and start typing then it take time and will started from next line issue only in chrome browser
https://www.loom.com/share/bae52e145acd4607816cbf012223a446

Browser  - Windows
Chrome  89.0.4389.90 (64bit)

**Issue - 2**
I'm facing the same issue so please make sure about the below comment.
https://github.com/buddyboss/buddyboss-platform/pull/2042#issuecomment-781881429

**Comment by jitendrabanjara1991** [comment]
Hello @mehulkaklotar 
cc - @chetansatasiya @sourabhmatolia 
**Issue 1** - I have fixed above issue that @sumitbuddyboss had assigned in the above comment. So I have added some code for that. So, Please review it. 
**Issue 2** - This is the default JS functionality. - [http://ichord.github.io/At.js/](http://ichord.github.io/At.js/)  - Check with second and third box. Because first one is the textarea. And in the News Feed section we use div.
Here is the video for the proof - [https://screencast-o-matic.com/watch/creZYEVV3aJ](https://screencast-o-matic.com/watch/creZYEVV3aJ)
Please check it and let me know if anything else.
Thanks.

**Review by mehulkaklotar — APPROVED** [review]
@sumitbuddyboss please test this and give feedback.

**Comment by sumitbuddyboss** [comment]
Hello @jitendrabanjara1991 

Fixed >
I have checked in the dev branch and the release issue is resolved now for proof please check the below video.

Video:
https://www.loom.com/share/76b74aad598a4aaeadf7283d5fbb7730

@chetansatasiya 
Ready for release 🚀

**Pull request merged**